uizabase/src/test/java/vn/uiza/restapi/uiza/UZServiceTest.java

```diff
@@ -0,0 +1,570 @@
+@PrepareForTest({ TextUtils.class })
+public class UZServiceTest {
+
+    private Gson gson;
+    private MockWebServer mockServer;
+    private UZService uzService;
+
+    @Before
+    public void setup() {
+        PowerMockito.mockStatic(TextUtils.class);
+        gson = new Gson();
+        mockServer = new MockWebServer();
+        String mockServerBaseURL = mockServer.url(FakeData.BASE_URL).toString();
+        RestClient.init(mockServerBaseURL, FakeData.TOKEN);
+        uzService = RestClient.createService(UZService.class);
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        mockServer.shutdown();
+    }
+
+    /////////////////////////////////////
+    ///////////////// USER-MANAGEMENT ///
+    /////////////////////////////////////
+    @Test
+    public void retrieveAnUser_Success() {
+        String responseString =
+                getJsonFromResource(FakeData.USER_MANAGEMENT_PATH, "retrieveAnUser_success.json");
+        MockResponse response = new MockResponse().setResponseCode(200).setBody(responseString);
+        mockServer.enqueue(response);
+
+        TestSubscriber<Object> testSubscriber = new TestSubscriber<>();
+        uzService.retrieveAnUser(API_VERSION, "508d46b7-afa7-45e7-8141-fdeafe32a4cc")
+                .subscribe(testSubscriber);
+        testSubscriber.awaitTerminalEvent(2, TimeUnit.SECONDS);
+
+        testSubscriber.assertCompleted();
+        testSubscriber.assertNoErrors();
+
+        Object expect = gson.fromJson(responseString, JsonObject.class);
+        Object actual = gson.toJsonTree(testSubscriber.getOnNextEvents().get(0));
+        assertEquals(expect, actual);
+    }
+
+    @Test
+    public void retrieveAnUser_Failed_ClientError() throws Exception {
+        String error400Str = getJsonFromResource(FakeData.ERROR_PATH, "error_400.json");
+
+        MockResponse response = new MockResponse().setResponseCode(400).setBody(error400Str);
+        mockServer.enqueue(response);
+
+        TestSubscriber<Object> testSubscriber = new TestSubscriber<>();
+        uzService.retrieveAnUser(API_VERSION, "508d46b7-afa7-45e7-8141-fdeafe32a4cc")
+                .subscribe(testSubscriber);
+        testSubscriber.awaitTerminalEvent(2, TimeUnit.SECONDS);
+
+        assertTrue(testSubscriber.getOnErrorEvents().get(0) instanceof HttpException);
+        HttpException error = (HttpException) testSubscriber.getOnErrorEvents().get(0);
+        String responseBody = error.response().errorBody().string();
+        ErrorBody errorBody = gson.fromJson(responseBody, ErrorBody.class);
+
+        testSubscriber.assertNotCompleted();
+        testSubscriber.assertError(error);
+        assertEquals(errorBody.getCode(), 400);
+        assertNotNull(errorBody.getData());
+        assertEquals("BAD_REQUEST", errorBody.getType());
+    }
+
+    @Test
+    public void retrieveAnUser_Failed_ServerError() {
+        MockResponse response = new MockResponse().setResponseCode(502);
```

Comment:
we need to make sure all 5xx are handled, not just 502. Should only test for 500, 502, 503, 504.